### DISCUSSION TITLE
Connect Border router to OVS bridge

### DISCUSSION THREAD

**Discussion by WoutervanHoof** [openthread/openthread#11042] [General]
Hi all, 
For a while now I have been trying to connect the wpan0 interface created by the OpenThread border router to a network bridge created by OpenVSwitch. I am quite new to both OpenThread and OVS, and unfortunately I cannot figure out the right order of operations.

I will quickly explain what I am doing right now: I run both the Thread border router and OVS on a raspberry pi. I access the raspberry pi over a WiFi hotspot from my laptop. This laptop also runs the Faucet SDN controller for OVS. 

When I do not use OVS, but only The thread border router, I can ping the devices in my Thread network from my laptop, no problem.
When I add the wpan0 interface as a port to OVS, I cannot even ping the Thread devices from another port attached to the OVS bridge.

I mainly have two questions: 
1) The Border router assigns some IPs to the wpan0 interface, namely some mesh local IPv6 IPs, one Link local IP and one Global IP. From https://wiki.archlinux.org/title/Network_bridge  and https://docs.openvswitch.org/en/latest/faq/issues/ I got the impression that I maybe should move the global IP to the br0 interface, is this correct?
2) To build the border router agent I run `INFRA_IF_NAME=wlan0 ./script/setup`. I have tried finding out what this INFRA_IF_NAME is used for exactly, but I cannot figure out if I maybe should run this script with `INFRA_IF_NAME=br0` or `INFRA_IF_NAME=ovs-system`. Could someone give me some clarification on what this variable is used for exactly?

Thanks a lot for any feedback/help, feel free to reach out for any clarification if anything is unclear!

**Comment by jwhui** [comment]
A Thread Border Router (TBR) needs to manage routing between the Thread interface and infrastructure network interface.

A TBR will automatically configure Unique Local Prefixes on both interfaces as necessary so that devices on one network can address devices on another network.

A TBR will also automatically configure routes on both interfaces so that packets sent by devices on one network are properly routed through the TBR to reach devices on the other network.

The `INFRA_IF_NAME` configures the network interface name for the infrastructure network. I assume that is `br0`. I do not have direct experience with OpenVSwitch, so I am guessing here.

The OTBR codelab may help to explain some of the concepts more: https://openthread.io/codelabs/openthread-border-router
